kong/plugins/key-auth/handler.lua

```diff
@@ -149,6 +150,15 @@ end
 function KeyAuthHandler:access(conf)
   KeyAuthHandler.super.access(self)
 
+  -- check if preflight request and wether it should be authenticated
+  if conf.authenticate_preflight == false and get_method() == "OPTIONS" then
```

Comment:
there are others already: https://github.com/Mashape/kong/blob/master/kong/plugins/cors/schema.lua#L24

if this gets merged into `master` I plan to create a PR against `next` that removes the comments, reverts this to `if not conf.authenticate_preflight and ...`, and implements the migration.